src/test/java/net/snowflake/client/jdbc/DatabaseMetaDataResultSetLatestIT.java

```diff
@@ -32,4 +35,26 @@ public void testGetObjectNotSupported() throws SQLException {
       }
     }
   }
+
+  /** Added in > 3.17.0 */
+  @Test
+  public void testObjectColumn() throws SQLException {
+    try (Connection connection = getConnection();
+        Statement statement = connection.createStatement()) {
+      statement.execute(
+          "CREATE OR REPLACE TABLE TABLEWITHOBJECTCOLUMN ("
+              + "    col3 OBJECT("
+              + "      str VARCHAR,"
+              + "      num NUMBER(38,0)"
+              + "      )"
+              + "   )");
+      DatabaseMetaData metaData = connection.getMetaData();
+      try (ResultSet resultSet =
+          metaData.getColumns(
+              connection.getCatalog(), connection.getSchema(), "TABLEWITHOBJECTCOLUMN", null)) {
+        assertTrue(resultSet.next());
```

Comment:
1. Can we test this metadata and not rows only?
2. Should we be able to find similar values using regular query (`SELECT * FROM TABLEWITHOBJECTCOLUMN`) and querying `resultSet.getColumnMetadata(1)`?

Reply:
1. Not really. We don't provide extraction of nested fields from metadata in `SnowflakeDatabaseMetaDataResultSet` currently. 
2. Nope, this is strictly connected to `connection.getMetaData();` which executes command `show JDBC:DatabaseMetaData.getColumns()` and then extracts info for specific table